presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -373,7 +373,7 @@ public synchronized void createTable(
         Action<TableAndMore> oldTableAction = tableActions.get(schemaTableName);
         TableAndMore tableAndMore = new TableAndMore(table, Optional.of(principalPrivileges), currentPath, Optional.empty(), ignoreExisting, statistics, statistics);
         if (oldTableAction == null) {
-            HdfsContext context = new HdfsContext(session, table.getDatabaseName(), table.getTableName());
+            HdfsContext context = new HdfsContext(session, table.getDatabaseName(), table.getTableName(), true, currentPath);
```

Comment:
This is incorrect. The `tablePath` field is expected to be set to the table path, and not the write path

Reply:
They are the same.  `currentPath` was constructed from table location handle,  changed to `table.getStorage().getLocation` to make it clear.